src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyExtensions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class CorsPolicyExtensions
+    {
+        private static string _WildcardSubdomain = "*.";
+
+        public static bool IsOriginAnAllowedSubdomain(this CorsPolicy policy, string origin)
+        {
+            if (policy.Origins.Contains(origin))
+            {
+                return true;
+            }
+            var originUri = new Uri(origin, UriKind.Absolute);
+            return policy.Origins
+                .Where(o => o.Contains(_WildcardSubdomain))
```

Comment:
I believe the `policy.Origins` are configured with domains that include the protocol, such as `http://example.com` therefore I opted for `Contains`. However if you are concerned that this might allow someone to configure the policy with `http://sub.*.domain.com`, which would get turned into a Uri like `http://sub.domain.com` then I could alter the logic here. I guess it could be changed to instead look for `://*.` so that the wildcard part is just after the protocol.

Yes will add the `StringComparison.Ordinal` argument.